src/creatures/combat/combat.cpp

```diff
@@ -405,6 +407,11 @@ ReturnValue Combat::canDoCombat(std::shared_ptr<Creature> attacker, std::shared_
 			}
 		}
 	}
+
+	if (!g_callbacks().checkCallback(EventCallback_t::creatureOnTargetCombat, &EventCallback::creatureOnTargetCombat, attacker, target)) {
```

Comment:
In fact, it is necessary to return the function directly as is done in the other event, you have to think of a way to return both